tests/integration/components/sl-textarea-test.js

```diff
@@ -0,0 +1,305 @@
+        'SMALL',
+        "Label's text-danger <small> is rendered"
+    );
+});
+
+test(
+    'If "label" property is not populated, "optional" and "required" elements are not rendered even if populated',
+    function( assert ) {
+        this.render( hbs`
+            {{#sl-textarea optional=true required=true}}
+            {{/sl-textarea}}
+        ` );
+
+        assert.strictEqual(
+            this.$( '>:first-child' ).find( 'label > .text-info' ).length,
+            0,
+            "Label's text-info is not rendered"
+        );
+
+        assert.strictEqual(
+            this.$( '>:first-child' ).find( 'label > .text-danger' ).length,
+            0,
+            "Label's text-danger is not rendered"
+        );
+    }
+);
+
+test( 'If "label" property is not populated, label element is not rendered', function( assert ) {
+    this.render( hbs`
+        {{#sl-textarea label=""}}
+        {{/sl-textarea}}
+    ` );
+    assert.strictEqual(
+        Ember.typeOf( this.$( '>:first-child' ).find( 'label' ).prop( 'for' ) ),
+        'undefined',
+        'Label element is not rendered'
+    );
+});
+
+test( 'If "label" property is populated, label element is rendered', function( assert ) {
+    this.set( 'label', 'test' );
+    this.render( hbs`
+        {{#sl-textarea label="test"}}
+        {{/sl-textarea}}
+    ` );
+
+    const label = this.$(
+        'label[for="' + this.$( '>:first-child' ).find( 'textarea' ).prop( 'id' ) + '"]'
+    );
+
+    assert.strictEqual(
+        label.length,
+        1,
+        'Label is present'
+    );
+
+    assert.strictEqual(
+        Ember.$.trim( label.text() ),
+        this.get( 'label' ),
+        'Label text is expected value'
+    );
+});
+
+test( 'If "label" property is populated, "for" attribute is expected value', function( assert ) {
+    this.set( 'label', 'Test Label' );
+    this.render( hbs`
+        {{#sl-textarea label=label}}
+        {{/sl-textarea}}
+    ` );
+
+    assert.strictEqual(
+        this.$( '>:first-child' ).find( 'label' ).prop( 'for' ),
+        this.$( '>:first-child' ).find( 'textarea' ).prop( 'id' ),
+        'Label "for" property matches textarea\'s "id" property'
+    );
+});
+
+test( '"selectionStart" is supported', function( assert ) {
+    this.render( hbs`
+        {{#sl-textarea selectionStart=2 value="testValue"}}
+        {{/sl-textarea}}
+    ` );
+    this.$( '>:first-child' ).find( 'textarea' ).get( 0 ).setSelectionRange( 2, 8 );
+
+    assert.strictEqual(
+        this.$( '>:first-child' ).find( 'textarea' ).prop( 'selectionStart' ),
+        2,
+        'Textarea "selectionStart" property is expected value'
+    );
+});
+
+test( '"selectionEnd" is supported', function( assert ) {
+    this.render( hbs`
+        {{#sl-textarea selectionEnd=8 value="testValue"}}
+        {{/sl-textarea}}
+    ` );
+
+    this.$( '>:first-child' ).find( 'textarea' ).get( 0 ).setSelectionRange( 2, 8 );
```

Comment:
aziz, your "this doesn't work" example might actually work.  In jsfiddle, the javascript pane is getting focus after render.  Since an element on the page has focus other than the textarea, it is my understanding that it could not have a selection.  Only the currently focused element may have selection.  This is why a simulated selection is required.